Applications/SlicerApp/Testing/Python/QtMixinTest.py

```diff
@@ -0,0 +1,122 @@
+import os
+import unittest
+from __main__ import vtk, qt, ctk, slicer
+from slicer.ScriptedLoadableModule import *
+import time
+
+#
+# QtMixinTest
+#
+
+class QtMixinTest(ScriptedLoadableModule):
+  def __init__(self, parent):
+    parent.title = "QtMixinTest" # TODO make this more human readable by adding spaces
+    parent.categories = ["Testing.TestCases"]
+    parent.dependencies = []
+    parent.contributors = ["Johan Andruejol (Kitware)"]
+    parent.helpText = """
+    This is a self test that tests the piping of two CLIs through python
+    """
+    parent.acknowledgementText = """""" # replace with organization, grant and thanks.
+    self.parent = parent
+
+    # Add this test to the SelfTest module's list for discovery when the module
+    # is created.  Since this module may be discovered before SelfTests itself,
+    # create the list if it doesn't already exist.
+    try:
+      slicer.selfTests
+    except AttributeError:
+      slicer.selfTests = {}
+    slicer.selfTests['QtMixinTest'] = self.runTest
+
+  def runTest(self):
+    tester = QtMixinTestTest()
+    tester.runTest()
+
+#
+# QtMixinTest
+#
+
+class QtMixinTestWidget(ScriptedLoadableModuleWidget, slicer.util.QtWidgetMixin):
+  def __init__(self, parent = None, auto_setup=True):
+    ScriptedLoadableModuleWidget.__init__(self, parent)
+    slicer.util.QtWidgetMixin.__init__(self)
+
+    if auto_setup:
+      self.setup()
+
+  def setup(self):
+    moduleName = 'QtMixinTest'
+    scriptedModulesPath = os.path.dirname(slicer.util.modulePath(moduleName))
+    path = os.path.join(scriptedModulesPath, 'Resources', 'UI', moduleName + '.ui')
+
+    self.Widget = self.loadUI(path)
+    self.layout.addWidget(self.Widget)
+
+#
+# QtMixinTestLogic
+#
+
+class QtMixinTestLogic(ScriptedLoadableModuleLogic):
+  def __init__(self):
+    ScriptedLoadableModuleLogic.__init__(self)
+
+#
+# TwoCLIsInARowTestLogic
+#
+
+class QtMixinTestTest(ScriptedLoadableModuleTest):
+
+  def runTest(self):
+    """Run as few or as many tests as needed here.
+    """
+    self.test_loadUI()
+    self.test_get()
+
+  def test_loadUI(self):
+    mixinWidget = QtMixinTestWidget(auto_setup=False)
```

Comment:
Since `QtMixinTestWidget` is never instantiated with `auto_setup=True`, would it make sense to simplify the code and remove this option ?